src/libraries/Common/src/Interop/Interop.HostPolicy.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static class HostPolicy
+    {
+#if TARGET_WINDOWS
+        private const CharSet HostpolicyCharSet = CharSet.Unicode;
+#else
+        private const CharSet HostpolicyCharSet = CharSet.Ansi;
+#endif
```

Comment:
Not sure it worth adding separate files for Windows and Unix instead of if-def

Reply:
I kinda have a feeling this is exactly what `CharSet.Auto` does. Not sure why it is not used unless there is some incompatibility in `UnmanagedFunctionPointer`. Also, what's the current policy on specifying `ExactSpelling` on P/Invoke signatures?

Reply:
@filipnavara https://docs.microsoft.com/en-us/dotnet/standard/native-interop/charset?view=netcore-3.1 looks like `CharSet.Auto` is Unicode for both Windows and Unix. And yes I shouldn't rename `UnmanagedFunctionPointer` delegates I guess...

Reply:
The table seems to agree with me (the ".NET Core 3.0 and later and Mono on Unix" part).

Reply:
Ah, indeed 🙂